### PR TITLE
Introduce Rake, environment, new database handling

### LINKED ISSUE #29: Use smarter way to find database location
With this we can make our program work in different environments. I mean to get rid of the current hard-coded `/tmp/` path in the web app.

### PR DESCRIPTION
**Pull request by janschill**
This introduces some new changes: 
**Rake** is the Ruby default task runner. Exactly what the `contro.rb` has done before. Commands are executed:

```bash
bundle exec rake app:server:development
```

Implemented:
- Start/stop development/production server
- Create database
- Inspect development database
- Drop database
- Seed database
- Generate .env file and create random secret
Needs implementation:
- Test
- Flagging of messages
- Use more native Ruby methods instead of shell commands

The databases are now configured in the `config/database.yml`. Here we have configuration for three databases, that we will probably be using later on.
As of now, Rake will generate three sqlite databases. But maybe we should be looking into using an in-memory sqlite database for testing, MySQL/Postgres what not for production and sqlite for development. Or just always sqlite 😂.

A class `AppEnvironment` is implemented. Application-wide can now be the current environment be accessed. It is set either by the `.env`-file or with Rake when a server is spin up.

I have not touched the API yet, I will do that the next days.

Will close #29,  #48, #49

### CODE REVIEW

**Comment by alevani** [comment]
Argh why do we need to over engineer everything like this :(

**Comment by janschill** [comment]
> Argh why do we need to over engineer everything like this :(

Over-engineer? You were complaining that stuff didn't work. So what I did is, use something that more reliable than an over-engineered `control.rb`. This just makes life easier to setup the project without wanting to spend too much time on it.

Also communicate your problems better please. You are not expressing any constructive feedback to me, your comment gives me a lot of room to misinterpret what you are trying to say …

I would love to argue any part that is too overwhelming or seems _over-engineering_.

File: app/db/seeder.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    password: BCrypt::Password.create('secret')
+  )
+end
+
+10.times do
+  followers.insert(
+    who_id: users.all.sample[:user_id],
+    whom_id: users.all.sample[:user_id]
+  )
+end
```
Comment (JeppeV) [inline app/db/seeder.rb]: It looks to me like the seeding allows for a user to follow himself?

File: app/db/seeder.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+require 'bcrypt'
+require 'faker'
+require 'sequel'
+require 'literate_randomizer'
+
+DB = Sequel.connect('sqlite://db/development.db')
```
Comment (JeppeV) [inline app/db/seeder.rb]: Looks like only the development db is seeded and not the test db, as indicated in the Rakefile

Comment (janschill) [reply app/db/seeder.rb]: Yes, very good point. We will have to look how we want to approach testing, but yes I agree. Seeding the test database will also be necessary.

File: app/db/seeder.rb (hunk 1 above)
Comment (janschill) [reply app/db/seeder.rb]: Yes, I actually what would happen in the frontend.

**Pull request merged**